mountpoint-s3/scripts/fs_bench.sh

```diff
@@ -65,122 +96,45 @@ read_bechmark () {
-  ## sequential write
-  job_name="sequential_write"
-  bench_file=${mount_dir}/${job_name}_${RANDOM}.dat
-  dd if=/dev/zero of=$bench_file bs=256k conv=fsync > ${results_dir}/${job_name}.txt 2>&1 &
-  # get the process ID
-  dd_pid=$!
-
-  sleep $runtime_seconds
-  # send USR1 signal to print the result
-  kill -USR1 ${dd_pid}
-  sleep 0.1
-  kill ${dd_pid}
-
-  throughput_value=$(awk '/copied/ {print $10}' ${results_dir}/${job_name}.txt)
-  unit=$(awk '/copied/ {print $11}' ${results_dir}/${job_name}.txt)
-  # convert unit to MiB/s
-  case "$unit" in
-    GB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000*1000*1000/1024/1024}")
-      ;;
-    MB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000*1000/1024/1024}")
-      ;;
-    kB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000/1024/1024}")
-      ;;
-  esac
-
-  json_data="{\"name\":\"$job_name\",\"value\":$throughput_value,\"unit\":\"MiB/s\"}"
-  echo $json_data | jq '.' | tee ${results_dir}/${job_name}.json
-
-  # clean up the data file and the raw output file
-  sleep 10
-  rm $bench_file ${results_dir}/${job_name}.txt
-
-
-  ## sequential write with direct IO
-  job_name="sequential_write_direct_io"
-  bench_file=${mount_dir}/${job_name}_${RANDOM}.dat
-  dd if=/dev/zero of=$bench_file bs=256k oflag=direct conv=fsync > ${results_dir}/${job_name}.txt 2>&1 &
-  # get the process ID
-  dd_pid=$!
-
-  sleep $runtime_seconds
-  # send USR1 signal to print the result
-  kill -USR1 ${dd_pid}
-  sleep 0.1
-  kill ${dd_pid}
-
-  throughput_value=$(awk '/copied/ {print $10}' ${results_dir}/${job_name}.txt)
-  unit=$(awk '/copied/ {print $11}' ${results_dir}/${job_name}.txt)
-  # convert unit to MiB/s
-  case "$unit" in
-    GB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000*1000*1000/1024/1024}")
-      ;;
-    MB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000*1000/1024/1024}")
-      ;;
-    kB/s)
-      throughput_value=$(awk "BEGIN {print $throughput_value*1000/1024/1024}")
-      ;;
-  esac
-
-  json_data="{\"name\":\"$job_name\",\"value\":$throughput_value,\"unit\":\"MiB/s\"}"
-  echo $json_data | jq '.' | tee ${results_dir}/${job_name}.json
-
-  # clean up the data file and the raw output file
-  sleep 10
-  rm $bench_file ${results_dir}/${job_name}.txt
-
-  # unmount file system
-  sudo umount ${mount_dir}
-
-  # cleanup mount directory
-  rm -rf ${mount_dir}
+  jobs_dir=mountpoint-s3/scripts/fio/write
+
+  for job_file in "${jobs_dir}"/*.fio; do
+    # mount file system
+    mount_dir=$(mktemp -d /tmp/fio-XXXXXXXXXXXX)
+    cargo run --release ${S3_BUCKET_NAME} ${mount_dir} \
+        --allow-delete \
+        --prefix=${S3_BUCKET_TEST_PREFIX} \
+        --max-threads=${max_threads}
+    mount_status=$?
+    if [ $mount_status -ne 0 ]; then
+        echo "Failed to mount file system"
+        exit 1
+    fi
+
+    # set bench file
+    bench_file=${mount_dir}/${job_name}_${RANDOM}.dat
```

Comment:
Is this going to be annoying if we want to do a four thread version (since they need to write to four different files)? Maybe that's just a problem for future us.

Reply:
It will be a bit annoying. The reason we use random file names is to be able to run multiple benchmarks at the same time. For four thread version, I think instead of using random file names we can create a random directory instead and make some changes in the `run_fio_job` function.